File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    
    let rest = lost.sort((a, b) => a - b);
    // 분실했는데 여분의 체육복이 있는 학생 제외 
    rest = lost.filter(l => {
        if(reserve.includes(l)){
            const index = reserve.indexOf(l);
            reserve.splice(index, 1);
            
            return false;
        } else return true;
    });
    
    // 빌려야 할 학생이 없는 경우
    if(rest.length === 0) return n;
    
    
    // 빌려야 할 학생이 있는 경우
    rest = rest.filter(r => {
        if(reserve.includes(r - 1) ){
            const index = reserve.indexOf(r - 1);
            
            reserve.splice(index, 1)
            
            return false;
        }  else if (reserve.includes(r + 1)) {
            const index = reserve.indexOf(r + 1);
            
            reserve.splice(index, 1)
        } else {
            return true;
        }
    })
    
    if(rest.length === 0) return n;
    else return n - rest.length;
    console.log("실제 빌려야 하는 학생", rest);
    
}

File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    print("도난 학생", lost)
    print("여벌 학생", reserve)
    
    pure_reserve = set(reserve) - set(lost)
    pure_lost = set(lost) - set(reserve)
    print("순수 여벌", pure_reserve)
    print("순수 도난", pure_lost)
    
    for r in pure_reserve:
        if r - 1 in pure_lost:
            pure_lost.remove(r - 1)
        elif r + 1 in pure_lost:
            pure_lost.remove(r + 1)
        
    attendences = n - len(pure_lost)
    
    print("참석 학생 수", attendences)
    
    return attendences